src/gettext.cpp

```diff
@@ -515,4 +515,17 @@ std::string strftime(const std::string& format, const std::tm* time)
 	return dummy.str();
 }
 
+bool ci_search(const std::string& s1, const std::string& s2)
+{
+	std::lock_guard<std::mutex> lock(get_mutex());
+
+	const std::locale& locale = get_manager().get_locale();
+
+	std::string ls1 = bl::to_lower(s1, locale);
+	std::string ls2 = bl::to_lower(s2, locale);
+
+	return std::search(ls1.begin(), ls1.end(),
+					   ls2.begin(), ls2.end()) != ls1.end();
+}
```

Comment:
I assume that unit_create.cpp and unit_recruit.cpp (#3787) should start using this version of ci_search? Also @rocketBANG's #3759 once it's merged?

Reply:
the line above also seems to have wrong indentation.

Reply:
@jostephd 
Yes, in these filters case-sensitive search for the Russian language.
For example, in the hotkeys preferences:
![screenshot_20190107_081121](https://user-images.githubusercontent.com/46393722/50743748-80dd8780-1256-11e9-9fc2-ae8ad28644c1.png)
![screenshot_20190107_081153](https://user-images.githubusercontent.com/46393722/50743754-9baffc00-1256-11e9-8946-f24beb2cca69.png)

@gfgtdf Done

Reply:
Okay. I assume those filters should be case-insensitive for Russia. Could you fix those other places to use translation::ci_search instead of their own versions of ci_search?

Reply:
[Done](https://github.com/wesnoth/wesnoth/pull/3847/commits/c6f4d72330870fa42459a66b0742501182fa5c03)